service/src/main/java/uk/gov/hmcts/reform/fpl/validators/HasTelephoneOrMobileValidator.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.hmcts.reform.fpl.validators;
+
+import org.apache.commons.lang.StringUtils;
+import uk.gov.hmcts.reform.fpl.model.ApplicantParty;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasTelephoneOrMobile;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class HasTelephoneOrMobileValidator implements ConstraintValidator<HasTelephoneOrMobile, ApplicantParty> {
```

Comment:
I asked to make it generic. Use `Party` class instead of `ApplicantParty`.

Reply:
Do not forget to remove applicant word from variable names.

Reply:
Party class does not have 'getMobileNumber' this belongs to ApplicantParty